firefox-ios/Client/Frontend/Browser/DownloadQueue.swift

```diff
@@ -283,6 +319,20 @@ class DownloadQueue {
             downloadProgress[uuid] = nil
         }
 
+        if let hasContentEncoding = download.hasContentEncoding,
+           hasContentEncoding,
+           let download = download as? HTTPDownload {
+            // We need to do a quick HEAD request to the server to check the uncompressed file size. Otherwise, the download
+            // progress will appear to exceed the `totalBytesExpected`. [FXIOS-9039]
+            // This is because NSURLSession automatically decodes gzip/compress encoded content and reports the decoded
+            // bytes to the delegate methods.
+            Task {
+                if let size = await download.getUncompressedSize(), size != download.totalBytesExpected {
+                    downloadProgress[uuid]?.totalExpected = size
```

Comment:
@ih-codes I could be misremembering but I _think_ this line will cause problems with multiple downloads. It's been a while since I looked at this code but IIRC users can enqueue more than 1 download simultaneously per window, and the `totalExpected` value is just incremented for each download.

So IOW, if the user has 3 downloads going at once, this will actually clear out / reset the `totalExpected` size in the wrong way.

Reply:
Eek! Good catch, I never tested triggering multiple downloads. 🤔  If we go forward with this workaround I'll check this and see if more work is needed.